### PR TITLE
Debug tools for Antelope lib

### LINKED ISSUE #689: Debug tools for Antelope lib
## Overview
Since we manage all the data displayed on the site using this library, it is crucial to have a robust set of development tools for debugging the content and behavior of this library.

## Ideas (draft)
- incorporate a function on each store that prints on the console the current state of the store.
- populate the window object with useful functions that can executed from the console
  - access the stores by calling useStoreName()
  - print all store's data at once

### PR DESCRIPTION
**Pull request by Viterbo**
# Fixes #689 and #745

## Description
This PR includes changes for two different issues. First, it incorporates a whole new functionality on the Antlib to provide tools for debugging. Then, there are some code refactors to remove unnecessary dependencies.

## Antlib debug tools (#689)
This is the list of features implemented.
- The debug mode can be turned on (and off) using a URL parameter like this `trace=true` but now is remembered in localStorage. It remains on if you lose the parameter while you navigate.
- You can print the current state of the stores to see the actual data
- When debug mode is on, you can access Antlib from the console and execute any valid function

## Dependencies removed (#745)
All proposed changes here were made. I will explain how I replaced those dependencies.
- Chan: needs to query the current stake/unstake ratio and for that, it used Account and Contract stores. That was removed with a direct `ethers.Contract` creation using the `getAntelope().wallets.getWeb3Provider();` global function to solve the provider.
- Contract: was using the Account store to get the authenticator and provide a signer for the contracts. The code was copied from OBE's code but the reality is that we don't need them anymore because we solve all signing with the EVMAuthenticators and they use their own signer. So, although there's no need for this functionality anymore, I refactored all functions to receive a final optional parameter bringing an `ethers.Signer` if for some reason the developer wants to sign this way.
- EVM: was using the Account store to initialize an InjectedAuthenticator. Now that initialization occurs inside the Authenticator itself. So, the EVM store now does not need to depend on Account store anymore
- Account: was calling a clear function one by one on all its dependencies. That code was replaced with a new event called getAntelope().events.onClear that passes the label indicating which context we need to clear (thinking of the future when we have multiple connections, we don't want to clean them all). 

## Test scenarios
Let's check the new Library debugging features: 
 ### Debug mode (on & off)
- go to this link with the trace = true https://deploy-preview-690--wallet-staging.netlify.app/?trace=true
- open console (F12)
   - you should see the trace turned on and some traces
- Login
  - you should see all Antlib login related funcion being executed
  - You lost the trace=true parameter in the url
  - now press F5 with no trace in the URL
    - you should keep seeing all traces
- To turn off the debug mode by accessing: https://deploy-preview-690--wallet-staging.netlify.app/?trace=false
  - open the console
  - you should not see any trace

### Print:
- Turn on debug mode again: https://deploy-preview-690--wallet-staging.netlify.app/?trace=true
- Open console
- Enter the following text in the console: `ant.print()`
  - you should see an explorable tree of data on every store

### Filtering traces
- Turn on debug mode again: https://deploy-preview-690--wallet-staging.netlify.app/?trace=true
- Login
- Open and lean the console
  - You will see that the update-related functions are called every now and then, which is annoying for debugging.
- Enter the following text in the console: `ant.debug.filterStart()`
  - you will notice that the traces continue but don't repeat.
  - after you get everything filtered, enter the following text: `ant.debug.filterEnd()`
  - now you will get trace only for your actions

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages
-   [x] I have included all english text to the translation file and/or created a new issue with the required translations for the currently supported languages
-   [x] I have tested for mobile functionality and responsiveness
-   [x] I have added appropriate test coverage

### CODE REVIEW

File: src/boot/antelope.ts [hunk 1]
```diff
@@ -97,6 +97,12 @@ export default boot(({ app }) => {
     ant.config.setIndexerHealthThresholdSeconds(10);
     ant.config.setIndexerHealthCheckInterval(5000);
 
+    // We only allow debug mode if we are not in production or in a sensitive environment
+    const weAreNotInProduction = process.env.NODE_ENV !== 'production';
+    const weAreInLocalhost = document.location.hostname === 'localhost';
+    const weAreInNetlify = document.location.hostname.includes('netlify');
+    ant.config.debug.allowDebugMode(weAreNotInProduction || weAreInLocalhost || weAreInNetlify);
```
Comment (Viterbo) [inline src/boot/antelope.ts:104]: This is a condition to allow debug mode or not. This is a configuration outside the Antlib so we can put any condition.

File: src/antelope/config/AntelopeDebug.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
+
+
+// creamos una clase llamada AntelopeDebug que nos permita almacenar el estado de estar
```
Comment (ezra-sg) [inline src/antelope/config/AntelopeDebug.ts:3]: would you mind translating this and the comment at line 11 into english?

File: src/antelope/stores/chain.ts [hunk 3]
```diff
@@ -103,6 +104,8 @@ const newChainModel = (network: string, isNative: boolean): ChainModel => {
 export interface ChainState {
     // chains mapped by label
     __chains: { [label: Label]: ChainModel };
+    // notwork settings
```
Comment (ezra-sg) [inline src/antelope/stores/chain.ts:107]:
```suggestion
    // network settings
```

**Review by ezra-sg — DISMISSED** [review]
🔥  great job

**Review by ezra-sg — DISMISSED** [review]

File: test/jest/__tests__/antelope/stores/balances.spec.ts [hunk 4]
```diff
@@ -60,14 +60,12 @@ describe('Antelope Balance Store', () => {
     });
 
     describe('Initializing the store', () => {
-        test('should initialize the store', () => {
-            expect(useFeedbackStore).not.toHaveBeenCalled();
-            expect(FeedbackActions.setDebug).not.toHaveBeenCalled();
-
+        test('should subscribe to the onAccountChanged event', () => {
```
Comment (donnyquixotic) [inline test/jest/__tests__/antelope/stores/balances.spec.ts:63]: can you remove or comment out the unused import on line 6? `FeedbackActions`?

**Review by donnyquixotic — CHANGES_REQUESTED** [review]
translation & unused import

File: src/antelope/config/AntelopeDebug.ts (hunk 2 above)
Comment (Viterbo) [reply src/antelope/config/AntelopeDebug.ts]: I used that text to generate code. Is not useful anymore. 
Thanks!

**Review by donnyquixotic — APPROVED** [review]

**Pull request merged**